mentoring/programs/mentoring-lead.md

```diff
@@ -0,0 +1,58 @@
+# Internship Mentoring Lead Handbook
+
+## Overview
+
+The Mentoring Lead role is created to help upstream Kubernetes contributors to participates in various mentorship programs like [GSoC](https://github.com/kubernetes/community/blob/master/mentoring/programs/google-summer-of-code.md), [GSoD](https://github.com/kubernetes/community/blob/master/mentoring/programs/google-season-of-docs.md), [LFX Mentorship](https://github.com/kubernetes/community/blob/master/mentoring/programs/lfx-mentorship.md) and many more as an independent organisation or under Cloud Native Computing Foundation(CNCF). 
+
+## Responsibilities
+ The role lead should:
+
+- Keep a calendar of dates for each program (when to submit our project name, when to submit the project scope, when selecting period is, etc) throughout the   year and
+- Coordinate all submissions and participation in each one
+- Go to sig meetings and slack channels to proactively help with scoping and planning work / job descriptions
+- Help onboard interns to the k8s specific community items like coordinate a hello for them at the k8s monthly community meeting
+- Help off board interns to the k8s specific community items like demo at monthly community meeting
```

Comment:
It might be helpful to also include an explicit point that they would be the main point of contact (unless delegated otherwise) for applicants and mentees, and should respond to queries within a reasonable amount of time.

This can also be a good place to add that the lead is actively encouraged to delegate work and mentor shadows.

I want to call this out specifically because the mentoring lead would end up getting a lot of DMs during the application period. It gets overwhelming. Delegation becomes *very* important here.